### PR TITLE
Create Elixir documentation`

### LINKED ISSUE #326: Document Elixir support (currently in private beta)
Here is a short version:

The build process first fetches the dependencies with mix deps.get , then mix deps.compile . It then runs npm install in either the root of your repository or in CC_PHOENIX_ASSETS_DIR  if it's defined. 

Next step runs mix compile, then npm run deploy (in repo root or defined dir, same as before).

Before running the application, we run mix phx.digest (you can replace phx.digest by defining CC_PHOENIX_DIGEST_GOAL ) followed by mix ecto.migrate.

Finally, we start the application with mix phx.server  (you can replace phx.server  by defining CC_PHOENIX_SERVER_GOAL ). You can completely replace this step by another command entirely by defining CC_RUN_COMMAND .

### PR DESCRIPTION
**Pull request by vvenance**
Should close #326

### CODE REVIEW

File: runtimes/elixir/elixir.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+## Overview
+Currently, Clever Cloud supports Elixir based applications.
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]: Currently? That's weird, no?

File: runtimes/elixir/elixir.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Create an Elixir application and add a PostreSQL database
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]: Why talk about a PostgreSQL database?

File: runtimes/elixir/elixir.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+In the Clever Cloud web console:
+- under the organization of your choice, click new, then application. Select **Elixir**, then name your application and choose it's deployment zone.
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
- under the organization of your choice, click new, then application. Select **Elixir**, then name your application and choose its deployment zone.
```

File: runtimes/elixir/elixir.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Configure your Phoenix application
+
+### Mandatory configuration
+
+- Edit the file `config/prod.secret.exs` to replace `System.get_env("DATABASE_URL")` with `System.get_env("POSTGRESQL_ADDON_URI")`.
+
+- Generate a secret token with `$ mix phx.gen.secret`.
+
+- Know your Elixir version with `$ elixir -v`
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
- Get your Elixir version with `$ elixir -v`, you will need it later
```

File: runtimes/elixir/elixir.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## Set production environment variables
+
+In the envrionment variables section of your Elixir's application, set
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
In the envrionment variables section of your Elixir application, set
```

File: runtimes/elixir/elixir.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+- **SECRET_KEY_BASE** with the value obtained with `$ mix phx.gen.secret` previously
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
- **SECRET_KEY_BASE** to the value obtained with `$ mix phx.gen.secret` previously
```

File: runtimes/elixir/elixir.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+- **CC_ELIXIR_VERSION** with the value of `$ elixir -v` result. (Available values as of today are `1.8`, `1.9` or `1.10`)
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
- **CC_ELIXIR_VERSION** to the value of `$ elixir -v` result. (Available versions as of today are `1.8`, `1.9` or `1.10`)
```

File: runtimes/elixir/elixir.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+`git+ssh://git@push.clever-cloud.com/<your_app_id>.git`. Copy it in your clipboard.
+
+
+- Locally, under your code folder, type in `git init` to set up a new git repository or skip this step if you already have one.
+- Add the deploy URL with `git remote add <name> <your-git-deployment-url>`.
+- Add your files via `git add <files path>` and commit them via `git commit -m <your commit message>`.
+- Now push your application on Clever Cloud with `git push <name> master`
+
+
+## Build and deployment phases and custom configuration
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
## Build, deployment phases and custom configuration
```

File: runtimes/elixir/elixir.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+When we push our code to the Clever Cloud remote provided, the following commands are run
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]:
```suggestion
Once you push your code to the Clever Cloud remote provided, the following commands are run:
```

File: runtimes/elixir/elixir.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+```
+$ mix deps.get
+$ mix deps.compile
+$ npm install
+```
+These commands will compile your dependencies at the root of your project folder. If you want to use another folder, specify it via the environment variable **CC_PHOENIX_ASSETS_DIR**.
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]: Unless I'm mistaken, `CC_PHOENIX_ASSETS_DIR` is only used for `npm install`. To change the folder for the entire build / run process, you should use `APP_FOLDER` (valid for all application types).

File: runtimes/elixir/elixir.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+Then `$ mix compile` is run. If you want to override this behavior, you can set the environment variable **CC_MIX_BUILD_GOAL** to the value you desire.
+
+At this point, there is the command `$ npm run deploy`.
+
+Then `$ mix phx.digest` is run. You can override this one with the variable **CC_PHOENIX_DIGEST_GOAL**.
+
+Finally, `$ mix phx.server` is invoked, and as always, you can override this behavior, either with **CC_RUN_COMMAND** where you have to specify the full command, or **CC_PHOENIX_SERVER_GOAL** where it will be a mix task by default.
+
+And if you need to specify the timezone of your application, you can do it with the variable **TZ** set to the usual timezone format, for instance `Europe/Paris`.
```
Comment (urcadox) [inline runtimes/elixir/elixir.md]: I'm not sure why this was necessary but maybe this should be put as an aside, like prefixing it with `note: ` or something.

File: runtimes/elixir/elixir.md (hunk 4 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

File: runtimes/elixir/elixir.md (hunk 5 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

File: runtimes/elixir/elixir.md (hunk 10 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: That's what I meant, thank you for pointing out it is not clear

File: runtimes/elixir/elixir.md (hunk 11 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: wich part do you talk about ? the Time Zone part or the CC_RUN_COMMAND and CC_PHOENIX_SERVER_GOAL or both ?

File: runtimes/elixir/elixir.md (hunk 2 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: As everything is really phoenix oriented (hi, env var names), I writted it assuming we were deplying a Phenix app, wich needs a PG database. WDYT ?

File: runtimes/elixir/elixir.md (hunk 1 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: yeah

Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

File: runtimes/elixir/elixir.md (hunk 10 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

File: runtimes/elixir/elixir.md (hunk 2 above)
Comment (Keruspe) [reply runtimes/elixir/elixir.md]: Maybe you could mention that this is an example using a phoenix application and that there might be some nuances if the user isn't using phoenix (at least 10% of our elixir users run somehting else).

Comment (vvenance) [reply runtimes/elixir/elixir.md]: thank you for the approach, will update when I'll have answer on the other comment then we'll be able to close

File: runtimes/elixir/elixir.md (hunk 11 above)
Comment (urcadox) [reply runtimes/elixir/elixir.md]: I was talking about the timezone part

File: runtimes/elixir/elixir.md (hunk 2 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

File: runtimes/elixir/elixir.md (hunk 11 above)
Comment (vvenance) [reply runtimes/elixir/elixir.md]: ✅

**Comment by vvenance** [comment]
ping @urcadox, all the requested changes are applied

**Pull request merged**